Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -500,6 +524,133 @@ - (void)viewWillAppear:(BOOL)animated
                                                                                                                @"Short name for edit menu item to share contents of media message.")
                                                                                       action:shareSelector],
                                                            ];
+
+    [self ensureBlockStateIndicator];
+}
+
+- (void)setUserHasScrolled:(BOOL)userHasScrolled {
+    _userHasScrolled = userHasScrolled;
+    
+    [self ensureBlockStateIndicator];
+}
+
+- (void)ensureBlockStateIndicator
+{
+    // This method should be called rarely, so it's simplest to discard and
+    // rebuild the indicator view every time.
+    [self.blockStateIndicator removeFromSuperview];
+    self.blockStateIndicator = nil;
+    
+    if (self.userHasScrolled) {
+        return;
+    }
+
+    NSString *blockStateMessage = nil;
+    if ([self isBlockedContactConversation]) {
+        blockStateMessage = NSLocalizedString(@"MESSAGES_VIEW_CONTACT_BLOCKED",
+                                              @"Indicates that this 1:1 conversation has been blocked.");
+    } else {
+        int blockedGroupMemberCount = [self blockedGroupMemberCount];
+        if (blockedGroupMemberCount == 1) {
+            blockStateMessage = [NSString stringWithFormat:NSLocalizedString(@"MESSAGES_VIEW_GROUP_1_MEMBER_BLOCKED_FORMAT",
+                                                                             @"Indicates that a single member of this group has been blocked."),
+                                 blockedGroupMemberCount];
+        } else if (blockedGroupMemberCount > 1) {
+            blockStateMessage = [NSString stringWithFormat:NSLocalizedString(@"MESSAGES_VIEW_GROUP_N_MEMBERS_BLOCKED_FORMAT",
+                                                                             @"Indicates that some members of this group has been blocked. Embeds "
+                                                                             @"{{the number of blocked user in this group}}."),
+                                 blockedGroupMemberCount];
+        }
+    }
+    
+    if (blockStateMessage) {
+        UILabel *label = [UILabel new];
+        label.font = [UIFont ows_mediumFontWithSize:14.f];
+        label.text = blockStateMessage;
+        label.textColor = [UIColor whiteColor];
+        
+        UIView * blockStateIndicator = [UIView new];
+        blockStateIndicator.backgroundColor = [UIColor ows_signalBrandBlueColor];
```

Comment:
I think we want material blue. signalBrandBlueColor is a bit darker and works well for text. e.g. compare to the navbar background.